### ISSUE TITLE
Option for JS-only build

### ISSUE THREAD

**Issue by adborden** [18F/federalist#1923]
#### User Story

ONRR is moving from Jekyll to [Gatsby](https://www.gatsbyjs.org/) a JavaScript framework for building static websites. The project needs a build option that only builds JavaScript (via `npm run federalist`). If we remove the `_config.yml`, Federalist will still run `jekyll build` which will clobber our build directory (`_site`). So we'd like a way to disable the Jekyll build.

#### Acceptance Criteria

- [ ] I can, with one click, configure my Federalist site at initialization or after its been started to use a NPM-only build option.

_After evaluating, edit this part:_

#### Implementation outline (if higher than "low" effort):

- [x]  Add additional build option in https://github.com/18F/federalist-garden-build for this use case
- [x] run the [ONRR code](https://github.com/ONRR/doi-extractives-data/) through the build script to test
- [x] deploy the test code in the garden container for staging
- [x] update https://github.com/18F/federalist staging to have a the next option
- [x] try a build!
- [x] @wslack update federalist-docs to instruct users to drop built pages into _sites folder
- [x] @wslack  to submit a PR to federalist-docs explaining "script only" build

**Comment by adborden** [comment]
There is a pretty hacky work around. If your JS build outputs to a staging directory e.g. `build`, you can create a simple config for Jekyll that just processes `build` to `_site`.

```yaml
# _config.yml
source: build
```

That way Jekyll will just copy files from build to _site. Since the files are already built (no front matter, no markdown) Jekyll won't do any processing, just copy them over.

**Comment by wslack** [comment]
@adborden is this time sensitive or are y'all ok with the hacky workaround at present?

**Comment by wslack** [comment]
@adborden what happens if for funsies, you try to set the site to use the "static" build option? We're wondering if that will work since https://github.com/18F/federalist-garden-build/blob/staging/tasks/build.py#L128 runs for static builds.

**Comment by adborden** [comment]
Quoting Will Slack (2018-07-31 11:26:54)
> @adborden what happens if for funsies, you try to set the site to use the "static" build option? We're wondering if that will work since https://github.com/18F/federalist-garden-build/blob/staging/tasks/build.py#L128 runs for static builds.

From my reading of the code, we would build our JS project into `_site`
but then the static job will copy everything in the source directory
(the repo) into `_site` as well, possibly overwriting some files.

I will try it out.


--
Aaron D Borden
Human and hacker

**Comment by wslack** [comment]
hmmmm ok. And also - how time sensitive is this if it doesn't work?

**Comment by adborden** [comment]
I'll let @master12 from ONRR speak to the timeline.

**Comment by master12** [comment]
I currently have another build option I could use until we can disable the Jekyll build. So its not an ASAP need but would like to get something for this in the next month or 2.

**Comment by wslack** [comment]
kk will take this into account during backlog grooming on Tues

**Comment by adborden** [comment]
Update: I tried out the static build and it timed out. I believe its trying to copy node_modules left over from the build which is several hundred MB and unnecessary at this point.

I added a script to delete everything in the source dir before the static copy. It half worked but there is still more debugging to do.

Anyway, I think the conclusion is that the static build works no better than the Jekyll build. Both require a work around.
